axelrod/ecosystem.py

```diff
@@ -0,0 +1,38 @@
+class Ecosystem(object):
+    """Create an ecosystem based on the payoff matrix from an Axelrod tournament."""
+
+    def __init__(self, pmatrix):
+
+        self.pmatrix = pmatrix
+        self.nplayers = len(self.pmatrix)
+
+        # Population histories will be recorded in a nested list,
+        # with each internal list containing a list of populations
+        # for all strategies. The first list, representing the
+        # starting populations, should have all equal values.
+        self.populations = [[1.0 / self.nplayers for i in range(self.nplayers)]]
+
+    def reproduce(self, turns):
+
+        for iturn in range(turns):
+
+            plist = list(range(self.nplayers))
+            pops = self.populations[-1]
+
+            # The unit payoff for each player in this turn is a sum of the payoffs
+            # obtained from all other players scaled by the size of the opponent population.
+            payoffs = [sum([self.pmatrix[ip][jp] * pops[jp] for jp in plist]) for ip in plist]
+
+            # In our case lower payoffs are better, so fitness should increase when
+            # the payoff descreases. This is a simple linear function that fulfills
+            # this and still behaves nicely, but we could chooise more complicated forms.
+            pmax = 5 * self.nplayers
```

Comment:
> Cool, let's keep it as it is for now, perhaps we could open an issue (more as a bookmark than anything else) to look in to this more? (Perhaps throw an inline comment in here also)

No, I wast too hasty and this is in fact not what I wanted. I wanted the payoff from each round to be in in the range 0...pmax, but forgot the payoffs are already scaled by the opponents population, so the range will be pmax-5...pmax. I will simplify this.